File: JULY-2025/21-07-2025/p1957.ex
defmodule Solution do
@spec make_fancy_string(s :: String.t) :: String.t
def make_fancy_string(<<char, rest::bits>> = str) when byte_size(str) > 0 do
  process_bits(rest, char, 1, <<char>>)
end
def make_fancy_string(""), do: ""

defp process_bits(<<char, rest::bits>>, char, 2, acc) do
  process_bits(rest, char, 2, acc)
end
defp process_bits(<<char, rest::bits>>, char, count, acc) do
  process_bits(rest, char, count + 1, <<acc::bits, char>>)
end
defp process_bits(<<char, rest::bits>>, _prev, _count, acc) do
  process_bits(rest, char, 1, <<acc::bits, char>>)
end
defp process_bits(<<>>, _prev, _count, acc), do: acc
end

File: JULY-2025/21-07-2025/p1957.scala
object Solution {
    def makeFancyString(s: String): String = {
        if (s.length < 3) s
        else {
            val array = new Array[Char](s.length);
            array(0) = s(0)
            array(1) = s(1)
            val i = s.drop(2).foldLeft(2) {
                case (i, c) =>
                    if (array(i - 1) == c && array(i - 2) == c) i
                    else {
                        array(i) = c
                        i + 1
                    }
            }
            array.take(i).mkString
        }
    }
}